api/controllers/AttendanceController.ts

```diff
@@ -17,10 +18,10 @@ export class AttendanceController {
   }
 
   @Get('/:uuid')
-  async getAttendancesForEvent(@Param('uuid') uuid: Uuid,
+  async getAttendancesForEvent(@Params() params: UuidParam,
```

Comment:
@shravanhariharan2 thoughts on using `@Params()` vs `@Param('uuid')`? I think we should stick with the singular if possible because otherwise if we want to add multiple params, e.g. an endpoint that accepts a UUID and pagination, we'd need some `UuidAndPaginationParams` object and etc. for all combinations. But I don't see a way to validate `@Param('uuid')`. Maybe it's nbd?

Reply:
I definitely prefer singular `Param('uuid')` over `Params()` since that kind of injection is the main benefit of routing-controllers, but yeah it doesn't look like `Param('uuid')` can be validated with class-validator. Not sure if there's any other route we can take like having a global middleware that checks if a uuid is passed in and validates it that way? Though that seems a bit more convoluted. 

If nothing else looks promising then I think it's nbd either, better to have a proper form of API validation at the expense of potentially more interfaces than no validation at all.

Reply:
Additionaly, if we do make an encapsulation of all parts of each request, these parameter class will be used like this lmore down the line